### PR TITLE
chore: update to rsmt2d v0.7.0

### LINKED ISSUE #712: Use the new rsmt2d.TreeConstructorFn
The block constructor is being changed in https://github.com/celestiaorg/rsmt2d/issues/119 (should be included in https://github.com/celestiaorg/rsmt2d/pull/116), after that, we need to update to the newer version of rsmt2d along with updating our usage of that constructor, specifically in the extended data square nmt inner nodes [caching mechanism](https://github.com/celestiaorg/celestia-app/blob/70ef3499e96752517c86a3649b6dbb361544ee26/pkg/inclusion/nmt_caching.go#L98-L116).

blocked by https://github.com/celestiaorg/rsmt2d/issues/119 / https://github.com/celestiaorg/rsmt2d/pull/116

### LINKED ISSUE #754: Refactor to update to rsmt2d v0.7.0
When we update to the next breaking release of rsmt2d, we will also need to refactor some of the tooling here to maintain compatibility.

specifically we will have to update the [inclusion](https://github.com/celestiaorg/celestia-app/tree/main/pkg/inclusion) and [wrapper](https://github.com/celestiaorg/celestia-app/tree/main/pkg/wrapper) packages. 

ref #712 https://github.com/celestiaorg/rsmt2d/pull/126

### PR DESCRIPTION
**Pull request by rahulghangas**
Updates dependency of rsmt2d to v0.7.0
Changes relevant constructors/methods to update to new definition of rsmt2d.TreeConstructorFn

- [x] Closes #712
- [x] Closes #754

### CODE REVIEW

**Comment by evan-forbes** [comment]
I'm working on fixing the remaining tests, as there are non-intuitive things in the wrapper (that we should eventually get rid of...)

**Review by evan-forbes — APPROVED** [review]
I'm approving, but we should probably get someone other than @rahulghangas or myself before merging

File: pkg/da/data_availability_header.go [hunk 1]
```diff
@@ -67,8 +67,8 @@ func ExtendShares(squareSize uint64, shares [][]byte) (*rsmt2d.ExtendedDataSquar
 			squareSize*squareSize,
 		)
 	}
-	tree := wrapper.NewErasuredNamespacedMerkleTree(squareSize)
-	return rsmt2d.ComputeExtendedDataSquare(shares, appconsts.DefaultCodec(), tree.Constructor)
+	// here we construct a tre
```
Comment (SweeXordious) [inline pkg/da/data_availability_header.go:70]:
```suggestion
	// here we construct a tree
```

Comment (SweeXordious) [reply pkg/da/data_availability_header.go:70]: Or even something like `tree construction`

**Review by SweeXordious — DISMISSED** [review]
Approving given the typo is fixed

**Review by SweeXordious — APPROVED** [review]

File: pkg/inclusion/nmt_caching_test.go [hunk 2]
```diff
@@ -154,13 +145,10 @@ func calculateSubTreeRoots(row [][]byte, depth int) [][]byte {
...
-		tr := wrapper.NewErasuredNamespacedMerkleTree(uint64(len(row)))
-		for j, r := range rowChunk {
-			c := (i * subLeafRange) + j
-			tr.Push(r, rsmt2d.SquareIndex{
-				Axis: uint(rsmt2d.Row),
-				Cell: uint(c),
-			})
+		tr := wrapper.NewErasuredNamespacedMerkleTree(uint64(len(row)), 0)
+		for _, r := range rowChunk {
+			// c := (i * subLeafRange) + j
```
Comment (rootulp) [inline pkg/inclusion/nmt_caching_test.go:150]: sorry for delayed review but this comment appears residual / safe to delete

**Pull request merged**